Print server name instead of ID on creation

srstevenson opened the pull request:
This depends on a change in the server client proposed in https://github.com/facultyai/faculty/pull/103/.

acroz commented:
@srstevenson and I agreed changed to the implementation in the discussion in facultyai/faculty#103 - holding off on review until then.

srstevenson commented:
I've updated this PR to make the second request to retrieve the server's name here in `faculty-cli`, rather than the server client, as discussed with @acroz. I've also removed the ID completely, rather than showing it in addition to the name. With both, the printed line is long and server name collision is unlikely with the autogenerated names, so it doesn't add much value.
